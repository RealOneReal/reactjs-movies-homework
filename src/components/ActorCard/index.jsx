import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import { truncateString } from "../../utils/stringUtils";

const ActorCard = ({ id, profile_path, name, character }) => {
  const handleClick = () => {
    console.log(id);
  };

  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const styles = {
    card: {
      width: "150px",
      height: "225px",
      cursor: "pointer",
      ":hover ": {
        transform: "translateY(-16px)",
        transition: ".3s ease",
      },
      bgcolor: "secondary.dark",
    },
    cardMedia: { height: "170px", borderRadius: "10px" },
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia component="img" image={profile_path} sx={styles.cardMedia} />
      <CardContent sx={{ p: "4px" }}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
