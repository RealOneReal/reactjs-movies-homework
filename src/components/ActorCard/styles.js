const styles = {
  card: {
    width: "150px",
    height: "225px",
    cursor: "pointer",
    ":hover ": {
      transform: "translateY(-16px)",
      transition: ".3s ease",
    },
    bgcolor: "secondary.dark",
  },
  cardMedia: { height: "170px", borderRadius: "10px" },
  cardContent: { p: "4px" },
};

export default styles;
