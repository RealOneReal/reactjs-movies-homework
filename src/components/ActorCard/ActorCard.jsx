import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import styles from "./styles";
import { truncateString } from "../../utils/stringUtils";
import Fallback from "../../assets/broken-image.png";

const ActorCard = ({ id, profile_path, name, character }) => {
  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const handleClick = () => {
    console.log(id);
  };

  const handleMediaFallback = (event) => {
    event.target.style.objectFit = "contain";
    return (event.target.src = Fallback);
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia
        component="img"
        alt={actorName}
        src={profile_path}
        sx={styles.cardMedia}
        onError={handleMediaFallback}
      />
      <CardContent sx={styles.cardContent}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
