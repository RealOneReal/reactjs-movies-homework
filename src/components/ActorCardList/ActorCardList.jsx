import { Grid } from "@mui/material";
import ActorCard from "../ActorCard";

const ActorCardList = ({ cast }) => {
  return (
    <Grid container spacing={3}>
      {cast.map((card) => (
        <Grid item key={card.id}>
          <ActorCard {...card} />
        </Grid>
      ))}
    </Grid>
  );
};

export default ActorCardList;
