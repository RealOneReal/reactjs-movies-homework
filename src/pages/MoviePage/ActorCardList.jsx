import { Grid } from "@mui/material";
import ActorCard from "../../components/ActorCard/ActorCard";
import { credits } from "../../mock-data/mockMovieDetail";

const ActorCardList = () => {
  const cast = credits.cast?.slice(0, 5);

  return (
    <Grid container spacing={3} sx={{ mt: 2 }}>
      {cast &&
        cast.map((card) => (
          <Grid item key={card.id}>
            <ActorCard {...card} />
          </Grid>
        ))}
    </Grid>
  );
};

export default ActorCardList;
