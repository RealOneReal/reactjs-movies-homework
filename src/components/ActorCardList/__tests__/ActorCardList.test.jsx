import { render } from "@testing-library/react";
import ActorCard from "../../ActorCard/ActorCard";
import ActorCardList from "../ActorCardList";

jest.mock("../../ActorCard/ActorCard", () => () => "ActorCard");

const mockData = {
  cast: [
    {
      name: "Leo",
      id: 1,
    },
    {
      name: "Kate",
      id: 2,
    },
    {
      name: "Mishka",
      id: 3,
    },
  ],
};

describe("ActorCardList", () => {
  it("should be rendered", () => {
    const { container } = render(<ActorCardList {...mockData} />);
    expect(container).toMatchSnapshot();
  });
});
