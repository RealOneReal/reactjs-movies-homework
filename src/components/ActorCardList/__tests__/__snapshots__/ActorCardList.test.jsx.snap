// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorCardList should be rendered 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      ActorCard
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      ActorCard
    </div>
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      ActorCard
    </div>
  </div>
</div>
`;
